gradle-plugin/kubernetes/src/main/java/org/eclipse/jkube/gradle/plugin/task/KubernetesResourceTask.java

```diff
@@ -16,10 +16,42 @@
 import javax.inject.Inject;
 
 import org.eclipse.jkube.gradle.plugin.KubernetesExtension;
+import org.eclipse.jkube.kit.common.util.LazyBuilder;
+import org.eclipse.jkube.kit.common.util.ResourceUtil;
+import org.eclipse.jkube.kit.config.resource.ResourceConfig;
+import org.eclipse.jkube.kit.config.service.JKubeServiceHub;
+import org.eclipse.jkube.kit.config.service.ResourceServiceConfig;
+import org.eclipse.jkube.kit.resource.service.DefaultResourceService;
+
+import java.io.File;
 
 @SuppressWarnings("CdiInjectionPointsInspection")
 public class KubernetesResourceTask extends AbstractJKubeTask {
 
+  @Override
+  protected JKubeServiceHub.JKubeServiceHubBuilder initJKubeServiceHubBuilder() {
+    JKubeServiceHub.JKubeServiceHubBuilder builder = super.initJKubeServiceHubBuilder();
+    realResourceDir = ResourceUtil.getFinalResourceDir(
+      kubernetesExtension.getResourceSourceDirectory().getOrElse(new File(javaProject.getBaseDirectory(), "src/main/jkube")),
+      kubernetesExtension.getResourceEnvironment().getOrNull()
+    );
```

Comment:
I the field is set here, it makes no sense to declare it in the parent class (https://github.com/eclipse/jkube/pull/853/files#r688239259)

Also, why not just creating a getApplicableResourceDirectory() in KubernetesExtension that performs the resolution logic?

Reply:
realResourceDir is required for extracting enricher config, I have only ported sections related to #835 . You can see this field is used in AbstractJKubeTask `extractEnricherConfig()` in ResourceTask PR #834

Reply:
I really don't get your point, if the field is used globally, it should be initialized in the parent class too, not only declared there.

Anyway, just follow the second suggestion:

> Also, why not just creating a getApplicableResourceDirectory() in KubernetesExtension that performs the resolution logic?